Server/EmotionDetection/redis_logger.py

```
@@ -33,6 +44,21 @@ def _connect(self):
 
+    def _ensure_connection(self):
+        """Ensure Redis connection is alive, reconnect if needed"""
+        try:
+            if self.redis_client:
+                self.redis_client.ping()
+                return True
+        except (redis.ConnectionError, redis.TimeoutError, AttributeError) as e:
+            logger.warning(f"Redis connection lost, reconnecting: {e}")
+            self._connect()
+            return self.redis_client is not None
+        except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)